src/main/java/org/prebid/server/bidder/ttx/TtxBidder.java

```diff
@@ -53,45 +56,56 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
     @Override
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
         final List<BidderError> errors = new ArrayList<>();
+        final MultiValuedMap<String, Imp> impsMap = new ArrayListValuedHashMap<>();
         final List<HttpRequest<BidRequest>> requests = new ArrayList<>();
 
         for (Imp imp : request.getImp()) {
             try {
                 validateImp(imp);
                 final ExtImpTtx extImpTtx = parseImpExt(imp);
                 final Imp updatedImp = updateImp(imp, extImpTtx);
-                requests.add(createRequest(request, updatedImp));
+                impsMap.put(computeKey(updatedImp), updatedImp);
             } catch (PreBidException e) {
                 errors.add(BidderError.badInput(e.getMessage()));
             }
         }
+
+        for (Collection<Imp> imps : impsMap.asMap().values()) {
+            requests.add(createRequest(request, List.copyOf(imps)));
```

Comment:
Each bidder obtains its own copy of `imps`.
See `org.prebid.server.auction.ExchangeService#prepareImps`.